### PR TITLE
Add put_file method

### PR DESCRIPTION
**Pull request by fdupont-redhat**
This pull request adds the ability to either create/override a file on the remote host, either from a file on local filesystem or from a variable. This will for example be used by V2V to create the throttling limits file on the conversion host.

Associated RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=1690851

### CODE REVIEW

**Comment by fdupont-redhat** [comment]
@miq-bot add_label transformation, enhancement, hammer/yes
@miq-bot add_reviewer @agrare 
@miq-bot add_reviewer @carbonin

File: lib/gems/pending/util/MiqSshUtil.rb [hunk 1]
```diff
@@ -40,13 +40,23 @@ def initialize(host, user, password, options = {})
...
-      $log.debug "MiqSshUtil::get_file - Copying of #{@host}:#{from} to #{to}, complete." if $log
+      $log&.debug("MiqSshUtil::get_file - Copying of #{@host}:#{from} to #{to}, complete.")
       return data
     end
   end
 
+  def put_file(to, content = nil, path = nil)
+    raise "Need to provide either content or path" if content.nil? && path.nil?
+    run_session do |ssh|
+      content ||= File.open(path, 'rb') { |f| f.read }
```
Comment (kbrock) [inline lib/gems/pending/util/MiqSshUtil.rb]: How is this different from `IO.read(path)` ?

Just a little curious.

(sorry, drive by PR comment)

Comment (fdupont-redhat) [reply lib/gems/pending/util/MiqSshUtil.rb]: Good question. It's a solution from stackoverflow... So, I'm ok with whatever is the preferred way.

Comment (fdupont-redhat) [reply lib/gems/pending/util/MiqSshUtil.rb]: Reading further, it seems that `IO.binread` has been introduced in Ruby 1.9.1, and should be preferred. So, I updated my code. Thanks for pointing this out.

**Comment by carbonin** [comment]
Removing the hammer label as the BZ is marked as an RFE and not labeled with a backport flag.

**Comment by carbonin** [comment]
@simaishi @dmetzger57 are we backporting RFEs?

**Comment by simaishi** [comment]
@carbonin We can backport RFEs if approved.

**Comment by fdupont-redhat** [comment]
It's a requirement for IMS 1.2, which is targeted on 5.10.3, so backporting would be great. Otherwise, I'll have to implement it in the ConversionHost class...

**Comment by carbonin** [comment]
Okay, I think that needs to be indicated on the BZ then, right?

**Comment by fdupont-redhat** [comment]
For other IMS 1.2 BZs, we simply set the right Target Release on the BZ and hammer/yes label on the PR. Though, I'm ok to comply with whatever you prefer. What key/value should I put in the BZ ?

**Comment by carbonin** [comment]
Okay, we should probably take this offline as it's a Red Hat specific process issue. I'll send an email.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 1834a57bbc5e2b16bd047e7f2d3d4846546e179d
Author: Nick Carboni <ncarboni@redhat.com>
Date:   Thu Mar 21 09:32:46 2019 -0400

    Merge pull request #426 from fdupont-redhat/add_put_file_method
    
    Add put_file method
    
    (cherry picked from commit 942c1243c081ff0b0165ae26b567126c1ff21e60)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1702085
```

**Pull request merged**